### ISSUE TITLE
initial data error

### ISSUE THREAD

**Issue by refda** [OpenSlides/openslides-manage-service#70]
Hello together!

I was installing the new version of OpenSlides. But after the command "./openslides initial-data" I was getting the following error:

`Error: setting initial data: setting initial data: rpc error: code = Unknown desc = inserting initial data into datastore: creating datastore objects: sending write request to datastore: got response `400 BAD REQUEST`: {"error":{"msg":"data.events[0] must be object","type":2,"type_verbose":"INVALID_REQUEST"}}`

The instance is runing but I don't have any admin password to login :/

Thanks for any tips.

**Comment by peb-adr** [comment]
I can confirm this, have had the exact same problem.
I then went ahead and tried bringing up the instance using older versions of the datastore. -> no difference
But when i used a locally built docker image from commit 97e57abf15eb8b558e153cc5ce8739f681e960ed it worked!
One commit later (6761e86c633bc91bbcb7836323096ec5cc84e188) the error starts occuring.
The problematic commit involves changes in `datastore.go` which i suspect to be the cause of this.
Also the problem seems to be on the server side of the manage-service. I tested older versions of the `openslides/manage` binary as well with no betterment.

In conclusion the latest working setup known to me is:
  - Latest `openslides` binary
  - Every openslides-service on latest version
  - Execpt `openslides-manage` built from 97e57abf15eb8b558e153cc5ce8739f681e960ed

So @normanjaeckel I think you should have another look at this and see if you can do anything about it ;)

**Comment by normanjaeckel** [comment]
Please post the request, that is send to the datastore.

**Comment by tanjix** [comment]
I also can confirm this behaviour, getting exactly the same here:

`root@openslides:/openslides/v4#  ./openslides initial-data
Error: setting initial data: setting initial data: rpc error: code = Unknown desc = inserting initial data into datastore: creating datastore objects: sending write request to datastore: got response `400 BAD REQUEST`: {"error":{"msg":"data.events[0] must be object","type":2,"type_verbose":"INVALID_REQUEST"}}`

**Comment by normanjaeckel** [comment]
See https://github.com/OpenSlides/openslides-manage-service/blob/main/pkg/datastore/datastore.go#L91 The "events" property contains an array/slice of objects.

**Comment by tanjix** [comment]
> 
> 
> See https://github.com/OpenSlides/openslides-manage-service/blob/main/pkg/datastore/datastore.go#L91 The "events" property contains an array/slice of objects.

Hi @normanjaeckel don't know, to whom the last comment was sent to, but at least for me, I don't know, what to do with it. I am end-user only and not being a developer with this deep knowledge of code. Thanks!

**Comment by r-peschke** [comment]
The interface of the datastore is defined [hier](https://github.com/OpenSlides/OpenSlides/wiki/Datastore-Service), see interface WriteRequest.
The interface is used intensive from the backend-service, which is written in Python. You could see a sample implementation beginning [with write](https://github.com/OpenSlides/openslides-backend/blob/d6e0497372290aec048a42860408162a48d380f1/openslides_backend/services/datastore/adapter.py#L459l) and finally send to datastore in [retrieve](https://github.com/OpenSlides/openslides-backend/blob/d6e0497372290aec048a42860408162a48d380f1/openslides_backend/services/datastore/adapter.py#L79).

**Comment by tanjix** [comment]
Great, waiting for further comments then, which maybe include a solution! Thanks, Guys, for your work on this! 👍

**Comment by tsiegleauq** [comment]
duplicate of #62 

@refda 
Furthermore: You are not really supposed to use OpenSlides 4 yet. Read this **carefully**.
https://github.com/OpenSlides/OpenSlides#using-openslides-productively

**Comment by normanjaeckel** [comment]
As talked with @emanuelschuetze : https://github.com/OpenSlides/openslides-manage-service/blob/main/pkg/datastore/datastore.go#L94 In this line we marshal a string to json.RawMessage. In this step we get extra quotation marks and therefor not a JSON object but a JSON string. Maybe something like `encodedEvent := []byte(event)` would help instead.

**Comment by ostcar** [comment]
You should not Marshall json.RawMessage. if it is valid JSON, than your can not Marshall it again. If it is no valid JSON, then it should not be json.RawMessage.

The solution is to remove the line at all and use event instead of encoded event later.

**Comment by peb-adr** [comment]
@normanjaeckel Yes thanks that fixed it :+1:
